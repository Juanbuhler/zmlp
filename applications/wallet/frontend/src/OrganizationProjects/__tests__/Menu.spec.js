import TestRenderer, { act } from 'react-test-renderer'

import OrganizationProjectsMenu from '../Menu'

const ORGANIZATION_ID = '42869703-fb62-4988-a0d1-e59b15caff06'
const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'

describe('<OrganizationProjectsMenu />', () => {
  it('should render properly', async () => {
    const mockFn = jest.fn()

    fetch.mockResponseOnce('{}')

    const component = TestRenderer.create(
      <OrganizationProjectsMenu
        organizationId={ORGANIZATION_ID}
        projectId={PROJECT_ID}
        revalidate={mockFn}
      />,
    )

    // Open Menu
    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Toggle Actions Menu' })
        .props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()

    // Select Delete Project
    act(() => {
      component.root.findByProps({ children: 'Delete Project' }).props.onClick()
    })

    // Cancel
    act(() => {
      component.root.findByProps({ children: 'Cancel' }).props.onClick()
    })

    // Open Menu
    act(() => {
      component.root
        .findByProps({ 'aria-label': 'Toggle Actions Menu' })
        .props.onClick()
    })

    // Select Delete Project
    act(() => {
      component.root.findByProps({ children: 'Delete Project' }).props.onClick()
    })

    // Confirm
    await act(async () => {
      component.root
        .findByProps({ children: 'Delete Permanently' })
        .props.onClick()
    })

    expect(fetch.mock.calls.length).toEqual(1)

    expect(fetch.mock.calls[0][0]).toEqual(`/api/v1/projects/${PROJECT_ID}/`)

    expect(fetch.mock.calls[0][1]).toEqual({
      headers: {
        'X-CSRFToken': 'CSRF_TOKEN',
        'Content-Type': 'application/json;charset=UTF-8',
      },
      method: 'DELETE',
    })

    expect(mockFn).toHaveBeenCalledWith()
  })
})
