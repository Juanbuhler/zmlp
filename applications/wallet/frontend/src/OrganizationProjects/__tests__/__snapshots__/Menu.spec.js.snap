// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrganizationProjectsMenu /> should render properly 1`] = `
.emotion-0 {
  position: relative;
  height: 100%;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 6px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #808080;
  background-color: transparent;
  margin: -6px;
}

.emotion-1:hover,
.emotion-1.focus-visible:focus {
  color: #ffffff!important;
}

.emotion-2 {
  position: absolute;
  z-index: 1;
  top: 100%;
  right: 0;
  background-color: #808080;
  border-radius: 2px;
  box-shadow: 0 4px 7px 0 #000000;
  min-width: 200px;
  font-family: Roboto,Avenir,"sans-serif";
}

.emotion-2 ul {
  list-style-type: none;
  padding: 0;
  padding-top: 4px;
  padding-bottom: 4px;
  margin: 0;
}

.emotion-2 li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 0;
  padding: 8px 16px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 400;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
}

.emotion-3:hover,
.emotion-3.focus-visible:focus,
.emotion-3 .noop {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #4a4a4a;
}

.emotion-4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-5 {
  -webkit-transform: rotate(135deg);
  -moz-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
  opacity: 0.5;
  margin-top: -2px;
  margin-right: -4px;
}

<div
  className="emotion-0"
>
  <button
    aria-label="Toggle Actions Menu"
    className="actions emotion-1"
    onBlur={[Function]}
    onClick={[Function]}
    type="button"
  >
    <svg
      data-file-name="kebab.svg"
      height={20}
    />
  </button>
  <div
    className="emotion-2"
  >
    <div>
      <ul>
        <li>
          <a
            className="emotion-3"
            href="/76917058-b147-4556-987a-0a0f11e46d9b"
            onBlur={[Function]}
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <div
              className="emotion-4"
            >
              View Project
              <svg
                className="emotion-5"
                data-file-name="back.svg"
                height={20}
              />
            </div>
          </a>
        </li>
        <li>
          <a
            className="emotion-3"
            href="/76917058-b147-4556-987a-0a0f11e46d9b/users"
            onBlur={[Function]}
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <div
              className="emotion-4"
            >
              Manage Users
              <svg
                className="emotion-5"
                data-file-name="back.svg"
                height={20}
              />
            </div>
          </a>
        </li>
        <li>
          <a
            className="emotion-3"
            href="/76917058-b147-4556-987a-0a0f11e46d9b/api-keys"
            onBlur={[Function]}
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <div
              className="emotion-4"
            >
              Manage API Keys
              <svg
                className="emotion-5"
                data-file-name="back.svg"
                height={20}
              />
            </div>
          </a>
        </li>
        <li>
          <button
            className="emotion-3"
            onBlur={[Function]}
            onClick={[Function]}
            type="button"
          >
            Delete Project
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
